packages/core/src/Utils/Analytics/analytics.ts

```diff
@@ -0,0 +1,50 @@
+interface AnalyticsEvent {
+    name: string;
+    properties: {
+        [key: string]: any;
+    };
+}
+
+let eventQueue: AnalyticsEvent[] = [];
+
+const setEvent = (event: AnalyticsEvent): void => {
+    eventQueue.push(event);
+    localStorage.setItem('pending_events', JSON.stringify(eventQueue));
+};
+
+const trackEvent = (event: AnalyticsEvent): void => {
+    if (window.rudderanalytics) {
+        window.rudderanalytics.track(event.name, event.properties);
+    } else {
+        setEvent(event);
+    }
+};
+
+const loadPendingEvents = (): void => {
```

Comment:
```suggestion
const loadPendingEvents = (): void => {
```
call this in so called `handleCachedEvents` function